# Cental Interval Examples

## Example 1: Estimate Mean life of light bulbs:
- Population SD = 1,000 hours
- Sample Size = 64
- Sample Mean = 7,500

### a) Construct 95% Confidence Interval for population mean life
- Since Population SD is given, can use z-statistics:

In [1]:
sigma <- 1000
n <- 64
sample.mean <- 7500
sample.sd <- sigma/sqrt(n) #125

# 95% Confidence Interval, z = 1.96:
z <- abs(qnorm((1-0.95)/2))
std.error <- sample.sd #125
margin.error <- z*std.error #244.9955

lower.interval <- sample.mean - margin.error
upper.interval <- sample.mean + margin.error
paste(lower.interval, " to ", upper.interval)

[1] "7255.00450193249  to  7744.99549806751"

#### Covert into a function

In [2]:
mean_ci <- function(pop.mean, sample.size, sample.mean, confidence=0.95){
    sigma <- pop.mean
    n <- sample.size
    sample.mean <- sample.mean
    sample.sd <- sigma/sqrt(n)

    confidence <- confidence
    z <- abs(qnorm((1-confidence)/2))
    std.error <- sample.sd 
    margin.error <- z*std.error

    lower.interval <- sample.mean - margin.error
    upper.interval <- sample.mean + margin.error
    paste(lower.interval, " to ", upper.interval)
}

In [3]:
mean_ci(1000, 64,7500)

[1] "7255.00450193249  to  7744.99549806751"

### b) Can the manufacturer claim 8,000 hours of mean life?
- The manufacturer cannot claim that because with 95% confidence, maximum life is 7745

### c) Construct 95% Confidence Interval for population mean life if Population SD is 800

In [4]:
mean_ci(800, 64, 7500, 0.95)

[1] "7304.00360154599  to  7695.99639845401"

## Example 2: Amount spent on food by a 15 costumer sample:
7.42, 6.29, 5.83, 6.5, 8.54, 9.51, 7.10, 6.80, 5.90, 4.89, 6.50, 5.52, 7.90, 8.30, 9.60

### a) Construct 95% Confidence Interval for population mean amount spent on lunch
- Since Population SD is NOT given, have to use t-statistics:

In [5]:
x <- c(7.42,6.29,5.83,6.5,8.54,9.51,7.10,6.80,5.90,4.89,6.50,5.52,7.90,8.30,9.60)

sample.mean <- mean(x)
n <- length(x)
sample.sd <- sd(x)

confidence <- 0.95
degree.freedom <- n - 1
t <- abs(qt((1-confidence)/2, degree.freedom)) #2.14

std.error <- sample.sd/sqrt(n) #0.366
margin.error <- t*(std.error) #0.778

lower.interval <- sample.mean - margin.error
upper.interval <- sample.mean + margin.error
paste(lower.interval, " to ", upper.interval)

[1] "6.32052904703734  to  7.892804286296"

- 95% confidence that money spent on food is between 6.32 and 7.89 dollars

#### Convert into a function:

In [6]:
mean_ci_t <- function(x, confidence=0.95){
    sample.mean <- mean(x)
    n <- length(x)
    sample.sd <- sd(x)

    confidence <- confidence
    degree.freedom <- n - 1
    t <- abs(qt((1-confidence)/2, degree.freedom)) 

    std.error <- sample.sd/sqrt(n)
    margin.error <- t*(std.error) 

    lower.interval <- sample.mean - margin.error
    upper.interval <- sample.mean + margin.error
    paste(lower.interval, " to ", upper.interval)
}

In [7]:
x <- c(7.42,6.29,5.83,6.5,8.54,9.51,7.10,6.80,5.90,4.89,6.50,5.52,7.90,8.30,9.60)
mean_ci_t(x)

[1] "6.32052904703734  to  7.892804286296"

## Example 3: In a sample of 500, 53% agree:

### a) Construct 95% Confidence Interval for population proportion that agree
- Sample Size (n)  = 500
- Sample prop (p) = 53% = 0.53


- Standard Error (SE) = Sqrt( ( p* (1-p)) / n )
- Margin of Error (ME) = z*SE

In [8]:
n <- 500
p <- 0.53

confidence <- 0.95
z <- abs(qnorm((1-confidence)/2))
std.error <- sqrt( (p*(1-p))/n )
margin.error <- z*std.error

lower.interval <- p - margin.error
upper.interval <- p + margin.error
paste(lower.interval, " to ", upper.interval)

[1] "0.486252831126234  to  0.573747168873766"

#### Convert into a function:

In [9]:
prop_ci <- function(prop, sample.size, confidence=0.95){
    n <- sample.size
    p <- prop

    confidence <- confidence
    z <- abs(qnorm((1-confidence)/2))
    std.error <- sqrt( (p*(1-p))/n )
    margin.error <- z*std.error

    lower.interval <- p - margin.error
    upper.interval <- p + margin.error
    paste(lower.interval, " to ", upper.interval)
}

In [10]:
n <- 500
p <- 0.53
prop_ci(p,n)

[1] "0.486252831126234  to  0.573747168873766"

### b) Repeat with Sample Size = 5,000 
- For a large sample size, confidence interval would be narrow

In [11]:
prop_ci(0.53, 5000)

[1] "0.516165930517488  to  0.543834069482512"

## Example 4: With SD = 10 and Margin of Error = 3:

### a) What sample size is needed for 99% Confidence?
- Population SD = sigma
- Margin of Error (ME)
- Confidence Interval = 90% (z = 2.575)
- ME = z*( sigma/Sqrt(n) )
- n = ( (z*sigma)/(ME) )^2

In [12]:
sigma <- 10
margin.error <- 3
z <- abs(qnorm((1-0.99)/2))

n <- ( (z*sigma)/margin.error )**2
n 

[1] 73.72107

#### Convert into a function:

In [13]:
mean_sample_size <- function(pop.sd, margin.error, confidence=0.95){
    
    sigma <- pop.sd
    margin.error <- margin.error
    confidence <- confidence
    
    z <- abs(qnorm((1-confidence)/2))

    n <- ( (z*sigma)/margin.error )**2
    n 
}

In [14]:
mean_sample_size(pop.sd=10, margin.error=3, confidence=0.99)

[1] 73.72107

### b) What sample size is needed for 95% Confidence?

In [15]:
mean_sample_size(pop.sd=10, margin.error=3, confidence=0.95)

[1] 42.68288

## Example 5: With Sample Size of 300, proportion of shoppers influenced by:
- Free shipping = 18% = 0.18
- Discounts = 13% = 0.13
- Reviews = 9% = 0.09

### a) Construct 95% Confidence Interval of population proportion that are influenced by free shipping:

In [16]:
p <- 0.18
n <- 300

z <- abs(qnorm((1-0.95)/2))
std.error <- sqrt( (p*(1-p))/n )
margin.error <- z*std.error

lower.interval <- p - margin.error
upper.interval <- p + margin.error
paste(lower.interval, " to ", upper.interval)

[1] "0.136525895756422  to  0.223474104243578"

#### Convert prop_ci() function made above:

In [17]:
prop_ci(prop=0.18, sample.size=300, confidence=0.95)

[1] "0.136525895756422  to  0.223474104243578"

### b) Construct 95% Confidence Interval of population proportion that are influenced by discounts:

In [18]:
prop_ci(prop=0.13, sample.size=300, confidence=0.95)

[1] "0.0919443831294028  to  0.168055616870597"

### c) Construct 95% Confidence Interval of population proportion that are influenced by reviews:

In [19]:
prop_ci(prop=0.09, sample.size=300, confidence=0.95)

[1] "0.0576160802550171  to  0.122383919744983"

### d) Determine Sample Size necessary to construct 95% Confidence Interval within +/- 0.02
- ME = Sqrt( (p*q)/n ) * z
- n = z^2*p*q/ME^2

In [20]:
p <- 0.18
confidence <- 0.95

z <- abs(qnorm((1-confidence)/2))

margin.error <- 0.02

n <- ( z^2 * p * (1-p) )/margin.error^2
n 

[1] 1417.498

#### Convert into a function:

In [21]:
prop_sample_size <- function(prop, confidence=0.95){
    p <- prop
    confidence <- confidence

    z <- abs(qnorm((1-confidence)/2))

    margin.error <- 0.02

    n <- ( z^2 * p * (1-p) )/margin.error^2
    n 
}

In [22]:
prop_sample_size(prop=0.18, confidence=0.95)

[1] 1417.498

In [23]:
prop_sample_size(prop=0.13, confidence=0.95)

[1] 1086.172

In [24]:
prop_sample_size(prop=0.09, confidence=0.95)

[1] 786.5387